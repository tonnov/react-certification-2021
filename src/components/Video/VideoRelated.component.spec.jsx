import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import VideoRelated from './VideoRelated.component';
import infoVideo from '../../mock/single-video.json';

jest.mock('react-router-dom', () => {
  return {
    Link: jest.fn(({ children }) => <div role="link">{children}</div>),
  };
});

describe('VideoRelated component', () => {
  const [vid] = infoVideo.items;

  // const component = render(<VideoRelated video={vid} />);
  // component.debug();

  beforeEach(() => {
    render(<VideoRelated video={vid} />);
  });

  it('should match snapshot', () => {
    const snap = renderer.create(<VideoRelated video={vid} />);
    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('Should render a link', () => {
    // render(<VideoRelated video={vid} />);
    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
  });

  it('Should render an img', () => {
    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
  });

  it('Should render a title (note)', () => {
    const note = screen.getByRole('note');
    expect(note).toBeInTheDocument();
  });
});
