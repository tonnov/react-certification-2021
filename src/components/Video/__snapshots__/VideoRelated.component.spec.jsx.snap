// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoRelated component should match snapshot 1`] = `
<div
  role="link"
>
  <div
    className="sc-gtsrHT fCeESP"
    role="listitem"
  >
    <img
      className="sc-dlnjwi ieKdUs"
      role="img"
      src="https://i.ytimg.com/vi/T_j60n1zgu0/mqdefault.jpg"
    />
    <div
      className="sc-hKFxyN bhYAoB"
    >
      <span
        className="sc-eCApnc hvrJiu"
        role="note"
      >
        Curso REACT JS ⚛️ - Aprende desde CERO 📈 Componentes, State, JSX (Tutorial Desde Cero en Español)
      </span>
      <span
        className="sc-jSFjdj itozLk"
      >
        midudev
      </span>
    </div>
  </div>
</div>
`;
